subworkflows/nf-scil/preproc_t1/main.nf

```diff
@@ -0,0 +1,58 @@
+// ** Importing modules from nf-scil ** //
+include { DENOISING_NLMEANS } from '../../../modules/nf-scil/denoising/nlmeans/main'
+include { PREPROC_N4 } from '../../../modules/nf-scil/preproc/n4/main'
+include { IMAGE_RESAMPLE } from '../../../modules/nf-scil/image/resample/main'
+include { BETCROP_ANTSBET } from '../../../modules/nf-scil/betcrop/antsbet/main'
+include { BETCROP_CROPVOLUME } from '../../../modules/nf-scil/betcrop/cropvolume/main'
+
+workflow PREPROC_T1 {
+
+    take:
+        ch_image           // channel: [ val(meta), [ image ] ]
+        ch_template        // channel: [ val(meta), [ template ] ]
+        ch_probability_map // channel: [ val(meta), [ probability_map ] ]
+        ch_mask_nlmeans    // channel: [ val(meta), [ mask ] ]            , optional
+        ch_ref_n4          // channel: [ val(meta), [ ref, ref_mask ] ]   , optional
+        ch_ref_resample    // channel: [ val(meta), [ ref ] ]             , optional
+        ch_mask_crop       // channel: [ val(meta), [ mask ] ]            , optional
+
+    main:
+
+        ch_versions = Channel.empty()
+
+        // ** Denoising ** //
+        ch_nlmeans = ch_image.join(ch_mask_nlmeans)
+        DENOISING_NLMEANS ( ch_nlmeans )
+        ch_versions = ch_versions.mix(DENOISING_NLMEANS.out.versions.first())
+
+        // ** N4 correction ** //
+        ch_N4 = DENOISING_NLMEANS.out.image.join(ch_ref_n4)
+        PREPROC_N4 ( ch_N4 )
+        ch_versions = ch_versions.mix(PREPROC_N4.out.versions.first())
+
+        // ** Resampling ** //
+        ch_resampling = PREPROC_N4.out.image.join(ch_ref_resample)
+        IMAGE_RESAMPLE ( ch_resampling )
+        ch_versions = ch_versions.mix(IMAGE_RESAMPLE.out.versions.first())
+
+        // ** Brain extraction ** //
+        ch_bet = IMAGE_RESAMPLE.out.image.join(ch_template).join(ch_probability_map)
+        BETCROP_ANTSBET ( ch_bet )
+        ch_versions = ch_versions.mix(BETCROP_ANTSBET.out.versions.first())
+
+        // ** crop ** //
+        ch_crop = BETCROP_ANTSBET.out.t1.join(ch_mask_crop)
+        BETCROP_CROPVOLUME ( ch_crop )
+        ch_versions = ch_versions.mix(BETCROP_CROPVOLUME.out.versions.first())
+
+    emit:
+        image_nlmeans   = DENOISING_NLMEANS.out.image     // channel: [ val(meta), [ image ] ]
```

Comment:
Put `t1` and `mask` as the final output channels, it'll separate them from the intermediary outputs. And at the top. `mask` is not listed here yet, it's the one from crop.